java/maven/build.xml

```diff
@@ -56,8 +56,8 @@
         <!-- we use jarjar to repackage simple json, to avoid clashes with 3rd party maven plugins possibly including it in their dependencies -->
         <taskdef name="jarjar" classname="org.pantsbuild.jarjar.JarJarTask" loaderref="lib.path.loader">
             <classpath>
-                <pathelement location="./external/asm-7.0.jar"/>
-                <pathelement location="./external/asm-commons-7.0.jar"/>
+                <pathelement location="./external/asm-9.4.jar"/>
```

Comment:
Question: asm is provided in platform/libs.asm too, but it seems that since its version is driven by jarjar, it's better to have a separate compile-time external here ?

Reply:
i am not sure actually. I was just thinking that the best approach with asm is likely to use the same version everywhere if possible (its sometimes shaded). It is a library which is known to be safe to update.

my thinking was: step 1, have everything use the same version, step 2 in future, check if we can remove the duplicates where it makes sense (if not we just leave it).

Reply:
@sdedic has now been deduplicated via #5751